src/services/phd-client.ts

```diff
@@ -0,0 +1,177 @@
+import axios from 'axios'
+import 'dotenv/config'
+import { Count, DefaultCrudRepository, Entity } from '@loopback/repository'
+
+import { Cache } from '../services/cache'
+
+const logger = require('./logger')
+
+const FALLBACK_WARNING = 'Data from Pocket HTTP DB not fetched. Falling back to fetching from MongoDB.'
+const FAILURE_ERROR = 'Data from Pocket HTTP DB not fetched. No fallback set so data fetch failed.'
+
+type ModelRef = new (...args: any[]) => any //eslint-disable-line
+interface ModelProps extends ModelRef {
+  definition?: { properties: { [key: string]: { required?: boolean } } }
+}
+
+interface FindParams<T extends Entity> {
+  path: string
+  model: ModelRef
+  fallback: DefaultCrudRepository<T, unknown>['find']
+  cacheKey?: string
+  cache?: Cache
+}
+
+interface FindOneParams<T extends Entity> {
+  path: string
+  id: string
+  model: ModelRef
+  fallback: DefaultCrudRepository<T, unknown>['findOne']
+  cache?: Cache
+}
+
+interface CountParams<T extends Entity> {
+  path: string
+  model: ModelRef
+  fallback: DefaultCrudRepository<T, unknown>['count']
+}
+
+// TODO - Unit tests for PHD Client - all cases and fallbacks
+
+/** The PHDClient fetches data from the Pocket HTTP DB, and falls back to fetching from the Loopback repositorites
+ * (which connect to MongoDB) if the fetch fails or the returned data is missing required fields. */
+class PHDClient {
+  private apiKey: string
+  private baseUrl: string
+
+  constructor() {
+    this.apiKey = process.env.PHD_API_KEY
+    this.baseUrl = process.env.PHD_BASE_URL
+  }
+
+  async find<T extends Entity>({
+    path,
+    model,
+    cache,
+    cacheKey = 'blockchains', // Currently .find only used to get blockchains
+    fallback,
+  }: FindParams<T>): Promise<T[]> {
+    const url = `${this.baseUrl}/${path}`
```

Comment:
What do you think about creating some sort of Enum for the routes/path of PHD? 

Example:
https://github.com/pokt-foundation/pocket-js/blob/8e71ab6caccd1aa43ee3b8532e791165ed30f471/packages/abstract-provider/src/routes.ts#L11-L17

Ideally, we want some sort of versioning on PHD as we have in the API (f.g. /v1) so it is easier to upgrade later.

This is just an enhancement, won't block the PR because of this. Let me know what you think!

Reply:
I think this is a good idea but it requires some fiddling with Typescript that I don't think is worth the lift given everything else going on. Going to suggest that if we do this we wait until after go-live.